src/flask/ctx.py

```diff
@@ -130,7 +130,15 @@ def add_header(response):
 
     .. versionadded:: 0.9
     """
+    top = _request_ctx_stack.top
+    if top is None:
+        raise RuntimeError(
+            "This decorator can only be used at local scopes "
+            "when a request context is on the stack. For instance within "
+            "view functions."
+        )
     _request_ctx_stack.top._after_request_functions.append(f)
```

Comment:
You can directly do: `top._after_request_functions.append(f)` since you already defined `top` in line 133. Just a minor suggestion.

Reply:
Thank you for good catch!